test/system/src/test/java/io/pravega/test/system/ReadWriteAndAutoScaleWithFailoverTest.java

```diff
@@ -100,6 +100,8 @@ public void setup() {
 
     @After
     public void tearDown() {
+        testState.stopReadFlag.set(true);
```

Comment:
This change seems to be unrelated. Are you trying to cover the case in which the test case doesn't stop the reader and writers, and execute tear down, e.g., when the test times out?